unix/moduselect.c

```diff
@@ -55,6 +55,11 @@ typedef struct _mp_obj_poll_t {
     unsigned short len;
     struct pollfd *entries;
     mp_obj_t *obj_map;
+    short iter_cnt;
+    short iter_idx;
+    int flags;
+    // callee-owned tuple
+    mp_obj_t ret_tuple;
```

Comment:
I was bit by inlining once, so don't feel like practicing it again. Here's it's also an "optimization", if ipoll is not used, tuple is not allocated. And overall, before playing with inlining and stuff, and I'd like to see a compacting GC implemented, that would require tightening up object refs vs object ptrs matters in many places, no need to complicate it more with adhoc optimizations IMHO.